app/commands/v2/put_content.rb

```diff
@@ -221,6 +222,14 @@ def update_content_item(content_item)
         content_item.save!
       end
 
+      def update_last_edited_at_if_needed(content_item, last_edited_at = nil)
+        if last_edited_at.nil? && %w(major minor).include?(payload[:update_type])
+          last_edited_at = Time.zone.now
```

Comment:
Something like this @dougdroper ?
```ruby
      def last_edited_at_unspecified?
        payload[:last_edited_at].nil?
      end

      def major_or_minor_update_type?
        %w(major minor).include?(payload[:update_type])
      end

      def update_last_edited_at_if_needed(content_item)
        last_edited_at = payload[:last_edited_at]

        if last_edited_at_unspecified? && major_or_minor_update_type?
          last_edited_at = Time.zone.now
        end

        content_item.update_attributes(last_edited_at: last_edited_at) if last_edited_at
      end
```

Pulling out last_edited_at from the payload in two methods does not seem great, but neither does passing it in just to do .nil?, so I'm not sure how to improve on this.